mod_modular_vanilla/hooks/config/character.nut

```diff
@@ -11,7 +16,35 @@
 	MV_DiversionDamageMult = 0.75
 });
 
+local original_getClone = ::Const.CharacterProperties.getClone;
 ::MSU.Table.merge(::Const.CharacterProperties, {
+	function getClone()
+	{
+		local ret = original_getClone();
+		ret.MV_MoraleCheckBraveryCallbacks = clone this.MV_MoraleCheckBraveryCallbacks;
+		ret.MV_ForbiddenMoraleStates = clone this.MV_ForbiddenMoraleStates;
+		return ret;
+	},
+
 	// Part of modularization of player_party.updateStrength
 	MV_StrengthMult = 1.0
+	/*
+	You push functions to this array during `skill.onUpdate` or `skill.onAfterUpdate`.
+	These are then used to modify the Bravery during actor.checkMorale.
+	The functions pushed look like this:
+		function( _change, _type )
+		_change is the _change parameter in `actor.checkMorale`.
+		_type is the _type parameter in `actor.checkMorale`.
+	The functions return a table with this signature:
+	{
+		Add = <integer>
+		Mult = <float>
+	}
+	// Add is then added to Bravery during actor.checkMorale
+	// whereas Mult is multiplied with the BraveryMult.
+	*/
+	MV_MoraleCheckBraveryCallbacks = [],
```

Comment:
Instead of this callback system, we can have a single `skill_container.MV_getMoraleCheckBravery( _change, _type )` event that is always run at the beginning of every `actor.checkMorale`. However, I imagine that a very small fraction of skills actually want to affect bravery in such a complex way, therefore, having a full-fledged skill_container event in every `checkMorale` seems inefficient.

What are your thoughts?

Reply:
It would be efficient if it was proper event system ;)

Reply:
> on skill added it goes through its methods and registers the ones starting with on for proper events,
A few years ago I implemented this entire thing in MSU and I benchmarked it a bit and realized that it is not really worth the performance improvement. Yes, it was faster, but in practical terms you don't notice any difference when playing the game. So, I thought it's not worth it.

You are probably right that I can add a new on* event here and it won't make a huge difference. But I don't mind this callback system either. I will think about it.